x-pack/legacy/plugins/siem/public/components/header_page/editable_title.test.tsx

```diff
@@ -0,0 +1,162 @@
+    wrapper.update();
+
+    expect(
+      wrapper
+        .find('[data-test-subj="editable-title-input-field"]')
+        .first()
+        .exists()
+    ).toBe(true);
+  });
+
+  test('it shows the submit button', () => {
+    const wrapper = mount(
+      <TestProviders>
+        <EditableTitle title="Test title" onSubmit={submitTitle} isLoading={false} />
+      </TestProviders>
+    );
+
+    wrapper.find('button[data-test-subj="editable-title-edit-icon"]').simulate('click');
+    wrapper.update();
+
+    expect(
+      wrapper
+        .find('[data-test-subj="editable-title-submit-btn"]')
+        .first()
+        .exists()
+    ).toBe(true);
+  });
+
+  test('it shows the cancel button', () => {
+    const wrapper = mount(
+      <TestProviders>
+        <EditableTitle title="Test title" onSubmit={submitTitle} isLoading={false} />
+      </TestProviders>
+    );
+
+    wrapper.find('button[data-test-subj="editable-title-edit-icon"]').simulate('click');
+    wrapper.update();
+
+    expect(
+      wrapper
+        .find('[data-test-subj="editable-title-cancel-btn"]')
+        .first()
+        .exists()
+    ).toBe(true);
+  });
+
+  test('it DOES NOT shows the edit icon when in edit mode', () => {
+    const wrapper = mount(
+      <TestProviders>
+        <EditableTitle title="Test title" onSubmit={submitTitle} isLoading={false} />
+      </TestProviders>
+    );
+
+    wrapper.find('button[data-test-subj="editable-title-edit-icon"]').simulate('click');
+    wrapper.update();
+
+    expect(
+      wrapper
+        .find('[data-test-subj="editable-title-edit-icon"]')
+        .first()
+        .exists()
+    ).toBe(false);
+  });
+
+  test('it switch to non edit mode when canceled', () => {
+    const wrapper = mount(
+      <TestProviders>
+        <EditableTitle title="Test title" onSubmit={submitTitle} isLoading={false} />
+      </TestProviders>
+    );
+
+    wrapper.find('button[data-test-subj="editable-title-edit-icon"]').simulate('click');
+    wrapper.update();
+    wrapper.find('button[data-test-subj="editable-title-cancel-btn"]').simulate('click');
+
+    expect(
+      wrapper
+        .find('[data-test-subj="editable-title-edit-icon"]')
+        .first()
+        .exists()
+    ).toBe(true);
+  });
+
+  test('it should change the title', () => {
+    const newTitle = 'new test title';
+
+    const wrapper = mount(
+      <TestProviders>
+        <EditableTitle title="Test title" onSubmit={submitTitle} isLoading={false} />
+      </TestProviders>
+    );
+
+    wrapper.find('button[data-test-subj="editable-title-edit-icon"]').simulate('click');
+    wrapper.update();
+
+    wrapper
+      .find('input[data-test-subj="editable-title-input-field"]')
+      .simulate('change', { target: { value: newTitle } });
+
+    wrapper.update();
+
+    expect(
+      wrapper.find('input[data-test-subj="editable-title-input-field"]').prop('value')
+    ).toEqual(newTitle);
+  });
+
```

Comment:
I did not test it but I think that should be a good case too
```suggestion
test('it should NOT change the title when cancel', () => {
    const newTitle = 'new test title';

    const wrapper = mount(
      <TestProviders>
        <EditableTitle title="Test title" onSubmit={submitTitle} isLoading={false} />
      </TestProviders>
    );

    wrapper.find('button[data-test-subj="editable-title-edit-icon"]').simulate('click');
    wrapper.update();

    wrapper
      .find('input[data-test-subj="editable-title-input-field"]')
      .simulate('change', { target: { value: newTitle } });

    wrapper.update();
    
    wrapper.find('button[data-test-subj="editable-title-cancel-btn"]').simulate('click');
    wrapper.update();
    
    expect(
      wrapper.find('input[data-test-subj="editable-title-input-field"]').prop('value')
    ).toEqual('Test title');
  });
```

Reply:
When you press cancel the input field disappears and it is not a part of `wrapper`. What if we check the `Title` component?

Reply:
Yes sorry, you are correct, we should do that 
```
expect(wrapper.find('h1[data-test-subj="header-page-title"').text()).toEqual('Test title');
```